### DISCUSSION TITLE
Get logs should allow topic filters

### DISCUSSION THREAD

**Discussion by raphael-verdier** [wevm/viem#752] [Idea / Feature Request]
In the getLogs function, topics cannot be passed and instead are deduced from a simplified event filter.

This reduces the possibilities of what can be done with the getLogs RPC call. Topic filters are a powerful tool to find specific logs. I think it should be something opened in the function. 

In my case, in an indexer project, I cannot use Viem because this feature is missing.

**Comment by frolic** [comment]
See also https://github.com/wevm/viem/discussions/287
